src/test/java/Pages/BasePage.java

```diff
@@ -27,4 +27,11 @@ public BasePage(WebDriver givenDriver){
     public WebElement findElement(By locator){
         return wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
     }
+
```

Comment:
I recommend: make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

```
    protected Actions actions;
    protected WebDriverWait wait;
    protected WebDriver driver;
```